dev_guide/builds/advanced_build_operations.adoc

```diff
@@ -262,3 +262,31 @@ whenever the first build is run and produces a new image with the binary
 artifact, the second build is automatically triggered to produce a runtime image
 that contains that artifact. Therefore, both builds behave as a single build
 with two stages.
+
+[[build-pruning]]
+== Build Pruning
+
+By default, builds that have completed their lifecycle are persisted indefinitely.
+You can limit the number of previous builds that are kept around by supplying a positive
+integer value for `successfulBuildsHistoryLimit` or `failedBuildsHistoryLimit`
+as shown in the below sample build configuration.
+
+[source,yaml]
+----
+apiVersion: "v1"
+kind: "BuildConfig"
+metadata:
+  name: "sample-build"
+spec:
+  successfulBuildsHistoryLimit: 2 <1>
+  failedBuildsHistoryLimit: 2 <2>
+----
+<1> `successfulBuildsHistoryLimit` will prune builds with a status of `completed`
+<2> `failedBuildsHistoryLimit` will prune builds with a status of `failed`, `cancelled`, or `error`
```

Comment:
"will retain up to 2 builds with a status of completed"
"will retain up to 2 builds with a status of failed, cancelled, or error"